pkg/sdk/databases.go

```diff
@@ -306,30 +360,39 @@ func (opts *AlterDatabaseOptions) validate() error {
 }
 
 type DatabaseSet struct {
-	DataRetentionTimeInDays    *int    `ddl:"parameter" sql:"DATA_RETENTION_TIME_IN_DAYS"`
-	MaxDataExtensionTimeInDays *int    `ddl:"parameter" sql:"MAX_DATA_EXTENSION_TIME_IN_DAYS"`
-	DefaultDDLCollation        *string `ddl:"parameter,single_quotes" sql:"DEFAULT_DDL_COLLATION"`
-	Comment                    *string `ddl:"parameter,single_quotes" sql:"COMMENT"`
+	DataRetentionTimeInDays    *int                     `ddl:"parameter" sql:"DATA_RETENTION_TIME_IN_DAYS"`
+	MaxDataExtensionTimeInDays *int                     `ddl:"parameter" sql:"MAX_DATA_EXTENSION_TIME_IN_DAYS"`
+	ExternalVolume             *AccountObjectIdentifier `ddl:"identifier,equals" sql:"EXTERNAL_VOLUME"`
+	Catalog                    *AccountObjectIdentifier `ddl:"identifier,equals" sql:"CATALOG"`
+	DefaultDDLCollation        *string                  `ddl:"parameter,single_quotes" sql:"DEFAULT_DDL_COLLATION"`
+	LogLevel                   *LogLevel                `ddl:"parameter,single_quotes" sql:"LOG_LEVEL"`
+	TraceLevel                 *TraceLevel              `ddl:"parameter,single_quotes" sql:"TRACE_LEVEL"`
+	Comment                    *string                  `ddl:"parameter,single_quotes" sql:"COMMENT"`
 }
 
 func (v *DatabaseSet) validate() error {
-	return nil
+	var errs []error
+	if v.ExternalVolume != nil && !ValidObjectIdentifier(v.ExternalVolume) {
+		errs = append(errs, errInvalidIdentifier("DatabaseSet", "ExternalVolume"))
+	}
+	if v.Catalog != nil && !ValidObjectIdentifier(v.Catalog) {
+		errs = append(errs, errInvalidIdentifier("DatabaseSet", "Catalog"))
+	}
+	return errors.Join(errs...)
```

Comment:
Shouldn't we validate empty struct here and in Unset?

Reply:
True, added.